### PR TITLE
[TENSOR MERGE] Remove context

### PR DESCRIPTION
**Pull request by cpuhrsch**

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: aten/src/THC/generic/THCTensor.cpp [hunk 1]
```diff
@@ -407,7 +407,7 @@ void THCTensor_(select)(THCState *state, THCTensor *self, THCTensor *src, int di
     THTensor_setSizeAtDim(self, d, self->size(d+1));
     THTensor_setStrideAtDim(self, d, self->stride(d+1));
   }
-  THTensor_resizeDim(self, self->dim_ - 1);
+  THTensor_resizeDim(self, self->sizes_.size() - 1);
```
Comment (gchanan) [inline aten/src/THC/generic/THCTensor.cpp]: this is an implementation detail really -- `how about self->dim() - 1`

File: aten/src/THC/THCTensor.cpp [hunk 2]
```diff
@@ -222,7 +222,7 @@ void THCTensor_squeeze1d(THCState *state, THCTensor *self, THCTensor *src, int d
       THTensor_setSizeAtDim(self, d, self->size(d+1));
       THTensor_setStrideAtDim(self, d, self->stride(d+1));
     }
-    THTensor_resizeDim(self, self->dim_ - 1);
+    THTensor_resizeDim(self, self->sizes_.size() - 1);
```
Comment (gchanan) [inline aten/src/THC/THCTensor.cpp]: `self->dim() - 1` ?

File: aten/src/TH/generic/THTensor.cpp [hunk 3]
```diff
@@ -413,7 +413,7 @@ void THTensor_(select)(THTensor *self, THTensor *src, int dimension, int64_t sli
     THTensor_setSizeAtDim(self, d, self->size(d+1));
     THTensor_setStrideAtDim(self, d, self->stride(d+1));
   }
-  THTensor_resizeDim(self, self->dim_ - 1);
+  THTensor_resizeDim(self, self->sizes_.size() - 1);
```
Comment (gchanan) [inline aten/src/TH/generic/THTensor.cpp]: self->dim() - 1 ?

File: aten/src/TH/generic/THTensor.cpp [hunk 4]
```diff
@@ -535,7 +535,7 @@ void THTensor_(squeeze1d)(THTensor *self, THTensor *src, int dimension)
       THTensor_setSizeAtDim(self, d, self->size(d+1));
       THTensor_setStrideAtDim(self, d, self->stride(d+1));
     }
-    THTensor_resizeDim(self, self->dim_ - 1);
+    THTensor_resizeDim(self, self->sizes_.size() - 1);
```
Comment (gchanan) [inline aten/src/TH/generic/THTensor.cpp]: self->dim() - 1 ?

File: aten/src/TH/THTensor.hpp [hunk 5]
```diff
@@ -51,11 +49,11 @@ struct THTensor
     // _dim() returns the "old" TH dimension view where no dimensions represents an empty tensor.
     // dim()  returns the ATen view of the dimensionality, i.e. 0-sized dimensions are supported.
     inline int64_t _dim() const {
-      return is_empty() ? 0 : dim_;
+      return is_empty() ? 0 : sizes_.size();
```
Comment (gchanan) [inline aten/src/TH/THTensor.hpp]: dim()

File: aten/src/TH/THTensor.hpp [hunk 6]
```diff
@@ -64,7 +62,7 @@ struct THTensor
 
     // represents that numel() == 0.
     inline bool is_empty() const {
-      for (int64_t i = 0; i < dim_; ++i) {
+      for (size_t i = 0; i < sizes_.size(); ++i) {
```
Comment (gchanan) [inline aten/src/TH/THTensor.hpp]: dim()

**Review by gchanan — DISMISSED** [review]

**Pull request merged**